test/jdk/com/sun/jdi/InterruptHangTest.java

```diff
@@ -28,93 +28,220 @@
+        }
+
+        // Create the debuggee interruptor thread if needed.
+        Thread interruptorThread;
+        if (preciseMode) {
+            interruptorThread =
+                DebuggeeWrapper.newThread(new PreciseInterruptor(Thread.currentThread()));
             interruptorThread.start();
-            try {
-                sync.wait();
-            } catch (InterruptedException ee) {
-                System.out.println("Debuggee interruptee: interrupted before starting loop");
-            }
+        } else if (aggressiveMode) {
+            interruptorThread =
+                DebuggeeWrapper.newThread(new AggressiveInterruptor(Thread.currentThread()));
+            interruptorThread.start();
+        } else {
+            interruptorThread = null; // we are in "remote" interrupt mode
         }
 
         // Debugger will keep stepping thru this loop
-        for (int ii = 0; ii < 200; ii++) {
-            answer++;
+        for (int ii = 0; ii < INTERRUPTS_EXPECTED; ii++) {
+            boolean wasInterrupted = false;
             try {
-                // Give other thread a chance to run
+                // Give other thread a chance to interrupt
                 Thread.sleep(100);
             } catch (InterruptedException ee) {
-                System.out.println("Debuggee interruptee: interrupted at iteration: "
-                                   + ii);
+                answer++;
+                wasInterrupted = true;
+                boolean isInterrupted = Thread.currentThread().isInterrupted();
+                System.out.println("Debuggee interruptee(" + ii + "): isInterrupted(" + isInterrupted + ")");
+                if (preciseMode) {
+                    // When Thread.sleep() is interrupted, the interrupted status of the thread
+                    // should remain cleared (since InterruptedException was thrown), and the
+                    // the next interrupt won't come in until after the sync.notify() below.
+                    // Note this is not true for the aggressive and remote modes since
+                    // interrupts can come in at any time, even while we are handling
+                    // an intrrupt.
+                    if (isInterrupted) {
+                        throw new RuntimeException("Thread is interrupted but shouldn't be.");
+                    }
+                    synchronized(InterruptHangTarg.sync) {
+                        // Let the interruptor thread know it can start interrupting again
+                        sync.notify();
+                    }
+                }
+            }
+            // Every Thread.sleep() call should get interrupted
+            if (!wasInterrupted) {
+                throw new RuntimeException("Thread was never interrupted during sleep: " + ii);
+            }
+        }
+
+        if (interruptorThread != null) {
+            synchronized(InterruptHangTarg.sync) {
```

Comment:
Nit: A space is missed after `synchronized` keyword.

Reply:
ok

Reply:
Actually there are quite a few cases of this in this test, and most are pre-existing. The test consistently does not use a space, and I see this is pretty common in other JDI tests. Do we have an official policy on this?

Reply:
One important rule is to keep the style consistent. So, I'd suggest to ignore my comment then.